mlchain/client/base.py

```diff
@@ -112,7 +112,7 @@ def __init__(self, api_key=None, api_address=None, serializer='msgpack',
             else:
                 api_address = mlchain.API_ADDRESS
         self.api_address = api_address
-        self.headers = headers or dict
+        self.headers = headers
```

Comment:
Why not check if `headers` is a `dict` right here and you won't have to check it for evey post and get call ?

Also, please raise some form of exception to notify the user that you are only allowed to parse in `dict` for `headers`